bin/repo

```diff
@@ -199,21 +202,75 @@ function handle_scope() {
+            project_type=$(jq_query ${CRUCIBLE_HOME}/config/repos.json --arg family ${family} --arg project_name ${project} '.[$family][] | select(.name == $project_name) | .type')
+            project_repository=$(jq_query ${CRUCIBLE_HOME}/config/repos.json --arg family ${family} --arg project_name ${project} '.[$family][] | select(.name == $project_name) | .repository')
+            project_primary_branch=$(jq_query ${CRUCIBLE_HOME}/config/repos.json --arg family ${family} --arg project_name ${project} '.[$family][] | select(.name == $project_name) | ."primary-branch"')
+            project_checkout_mode=$(jq_query ${CRUCIBLE_HOME}/config/repos.json --arg family ${family} --arg project_name ${project} '.[$family][] | select(.name == $project_name) | .checkout.mode')
+            project_checkout_target=$(jq_query ${CRUCIBLE_HOME}/config/repos.json --arg family ${family} --arg project_name ${project} '.[$family][] | select(.name == $project_name) | .checkout.target')
+
+            printf "${fmt_line}\n" "${family}" "${project}" "${project_type}" "${project_repository}" "${project_primary_branch}" "${project_checkout_mode}" "${project_checkout_target}"
+        done
+    done
+}
+
+function config_help() {
+    echo "Config Usage:"
+    echo "  repo config [<command> <command-specific-options>]"
+    echo
+    echo "The following command are supported"
+    echo
+    echo "show          |  Dump the configuration information in tabular form (default)"
+    echo
+}
 
 case "${command}" in
-    info)
-        fmt_line="%-15s  %-13s  %-16s  %-58s  %-17s  %-15s  %-13s  %-14s  %-13s"
-        printf "${fmt_line}\n" "Project Name" "Project Type" "Git Local Branch" "Git Remote URL" "Git Remote Branch" "Git Remote Name" "Local Changes" "Current Commit" "Updates"
+    info|details)
+        if [ -z "${1}" ]; then
+            scope="all"
+        else
+            scope=${1}
+            shift
+        fi
 
-        handle_scope info ${scope}
+        case "${command}" in
+            info)
+                fmt_line="%-15s  %-13s  %-16s  %-60s  %-17s  %-15s  %-13s  %-14s  %-13s"
+                printf "${fmt_line}\n" "Project Name" "Project Type" "Git Local Branch" "Git Remote URL" "Git Remote Branch" "Git Remote Name" "Local Changes" "Current Commit" "Updates"
+
+                handle_scope info ${scope}
+                ;;
+            details)
+                handle_scope details ${scope}
+                ;;
+        esac
         ;;
-    details)
-        handle_scope details ${scope}
+    config)
+        source "${CRUCIBLE_HOME}/bin/jqlib"
+
+        config_subcommand="show"
+        if [ -n "${1}" ]; then
+            config_subcommand=${1}
+            shift
+        fi
+
+        case "${config_subcommand}" in
+            show)
+                config_show
+                ;;
+            help)
+                config_help
+                ;;
+            *)
+                config_help
+                exit 1
+                ;;
```

Comment:
we can only show config now... I am just wondering if allowing the user to change the config in the future is a good idea. Only advanced users need to change the config, so they can change the json directly?

Reply:
Well, I think it is complicated.  I do think we will probably want the ability to switch from one version to the next at a minimum.  So say for example you install a controller on version 2024.3 and that is in your repos.json but then eventually you want to move to 2024.4 when it is available.  I think we need a command for that...something like:

```
crucible repo config set-version 2024.4
```

and it would ripple through and change all of the repos for you individually.  Similarly, once you do that you might find something that breaks your workflow and so you might want to go backwards:

```
crucible repo config set-version 2024.3
```

That is very easy and I don't think a "power user" action.

Beyond that, it very well be that other actions I had in mind are optional since they are possible to do by editing the config file directly.  However, it should be noted that scripting them with a nice interface does afford us the ability to do things like validate the inputs, check the return codes for the changes of individual settings, etc. that a manual edit wouldn't do -- basically it would give us greater ability to maintain the integrity of the config file.

Additionally, while I wrote a schema for this JSON file at the moment I'm not actually doing anything with it.  I figured I would add that later.  Validation of the file and the ability to revert (or something else) is the kind of thing we can do when the changes are scripted as opposed to when manual.  Another reason why I didn't implement validation yet is I need to figure out when / how often to validate -- I don't want to do it over and over again needlessly so I need to carefully determine where to add calls to that.